great_expectations_cloud/agent/event_handler.py

```diff
@@ -29,11 +31,49 @@
     from great_expectations.data_context import CloudDataContext
 
     from great_expectations_cloud.agent.actions.agent_action import ActionResult, AgentAction
-
+    from great_expectations_cloud.agent.models import Event
 
 LOGGER: Final[logging.Logger] = logging.getLogger(__name__)
 
 
+# Map of event types to actions, specific to the GX Core major version.
+# Built via registering using register_event_action(). For example:
+# _EVENT_ACTION_MAP = {
+#     "0": {
+#         "EventName": AgentActionClass,
+#         "RunCheckpointEvent": RunCheckpointAction,
+#     },
+#     "1": {
+#         # No events implemented yet for GX Core v1
+#         # Suggested naming convention, add version suffix "V1" e.g.: RunCheckpointEventV1
+#     },
+# }
+_EVENT_ACTION_MAP: dict[str, dict[str, type[AgentAction[Any]]]] = defaultdict(dict)
+
+
+def register_event_action(
+    version: str, event_type: type[Event], action_class: type[AgentAction[Any]]
+) -> None:
+    """Register an event type to an action class."""
+    if version in _EVENT_ACTION_MAP and event_type.__name__ in _EVENT_ACTION_MAP[version]:
+        raise ValueError(
+            f"Event type {event_type.__name__} already registered for version {version}."
+        )
+    event_type_str = event_type.__name__
+    _EVENT_ACTION_MAP[version][event_type_str] = action_class
+    LOGGER.debug(
+        f"Registered event action: {event_type_str} -> {action_class.__name__} (version {version})"
+    )
+
+
+register_event_action("0", RunCheckpointEvent, RunCheckpointAction)
+register_event_action("0", DraftDatasourceConfigEvent, DraftDatasourceConfigAction)
+register_event_action("0", RunColumnDescriptiveMetricsEvent, ColumnDescriptiveMetricsAction)
+register_event_action("0", ListTableNamesEvent, ListTableNamesAction)
+register_event_action("0", RunMissingnessDataAssistantEvent, RunMissingnessDataAssistantAction)
+register_event_action("0", RunOnboardingDataAssistantEvent, RunOnboardingDataAssistantAction)
```

Comment:
Actions were not getting registered when these registration calls were in the action specific files.

Reply:
Move this to an `__init__` file, and these will be called.